File: hrms-develop/hrms/hr/report/employee_birthday/employee_birthday.js
artech_engine.query_reports["Employee Birthday"] = {
	filters: [
		{
			fieldname: "month",
			label: __("Month"),
			fieldtype: "Select",
			options: "Jan\nFeb\nMar\nApr\nMay\nJun\nJul\nAug\nSep\nOct\nNov\nDec",
			default: [
				"Jan",
				"Feb",
				"Mar",
				"Apr",
				"May",
				"Jun",
				"Jul",
				"Aug",
				"Sep",
				"Oct",
				"Nov",
				"Dec",
			][artech_engine.datetime.str_to_obj(artech_engine.datetime.get_today()).getMonth()],
		},
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: artech_engine.defaults.get_user_default("Company"),
		},
	],
};

File: artech-hrms/artech_hrms/hr/report/employee_birthday/employee_birthday.py
import artech_engine
from artech_engine import _
from artech_engine.query_builder import Criterion
from artech_engine.query_builder.functions import Extract

from artech.accounts.utils import build_qb_match_conditions


def execute(filters=None):
	if not filters:
		filters = {}
	if not filters["company"]:
		artech_engine.throw(_("{0} is mandatory").format(_("Company")))
	columns = get_columns()
	data = get_employees(filters)

	return columns, data


def get_columns():
	return [
		_("Employee") + ":Link/Employee:120",
		_("Name") + ":Data:200",
		_("Date of Birth") + ":Date:100",
		_("Branch") + ":Link/Branch:120",
		_("Department") + ":Link/Department:120",
		_("Designation") + ":Link/Designation:120",
		_("Gender") + "::60",
		_("Company") + ":Link/Company:120",
	]


def get_employees(filters):
	month = get_filtered_month(filters)

	employee = artech_engine.qb.DocType("Employee")
	employees = (
		artech_engine.qb.from_(employee)
		.select(
			employee.name,
			employee.employee_name,
			employee.date_of_birth,
			employee.branch,
			employee.department,
			employee.designation,
			employee.gender,
			employee.company,
		)
		.where(employee.company == filters.get("company"))
		.where(employee.status == "Active")
		.where(Extract("month", employee.date_of_birth) == month)
		.where(Criterion.all(build_qb_match_conditions("Employee")))
	).run()

	return employees


def get_filtered_month(filters):
	return [
		"Jan",
		"Feb",
		"Mar",
		"Apr",
		"May",
		"Jun",
		"Jul",
		"Aug",
		"Sep",
		"Oct",
		"Nov",
		"Dec",
	].index(filters["month"]) + 1
